Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -531,9 +520,27 @@ bool HGCalDDDConstants::isValidHex8(int layer, int modU, int modV, int cellU, in
       }
     }
   }
+  return true;
+}
+
+bool HGCalDDDConstants::isValidHex8(int layer, int modU, int modV, int cellU, int cellV, bool fullAndPart) const {
+  // First check validity for a layer|wafer| of post TDR version
+  if (!isValidHex8(layer, modU, modV, fullAndPart))
+    return false;
+  int indx = HGCalWaferIndex::waferIndex(layer, modU, modV);
+  auto itr = hgpar_->typesInLayers_.find(indx);
```

Comment:
do you need to check for the case where indx is not found?

Reply:
indx is basically a packed id. It is a calculated number.